src/openrct2/actions/RideSetAppearanceAction.hpp

```diff
@@ -0,0 +1,182 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../Cheats.h"
+#include "../Context.h"
+#include "../core/MemoryStream.h"
+#include "../drawing/Drawing.h"
+#include "../interface/Window.h"
+#include "../localisation/Localisation.h"
+#include "../localisation/StringIds.h"
+#include "../ride/Ride.h"
+#include "../ui/UiContext.h"
+#include "../ui/WindowManager.h"
+#include "../world/Park.h"
+#include "GameAction.h"
+
+enum class RideSetAppearanceType : uint8_t
+{
+    TrackColourMain,
+    TrackColourAdditional,
+    TrackColourSupports,
+    MazeStyle = TrackColourSupports,
+    VehicleColourBody,
+    VehicleColourTrim,
+    VehicleColourTernary,
+    VehicleColourScheme,
+    EntranceStyle
+};
+
+DEFINE_GAME_ACTION(RideSetAppearanceAction, GAME_COMMAND_SET_RIDE_APPEARANCE, GameActionResult)
+{
+private:
+    NetworkRideId_t _rideIndex{ -1 };
+    RideSetAppearanceType _type;
+    uint8_t _value;
+    uint32_t _index;
+
+public:
+    RideSetAppearanceAction()
+    {
+    }
+    RideSetAppearanceAction(ride_id_t rideIndex, RideSetAppearanceType type, uint8_t value, uint32_t index)
+        : _rideIndex(rideIndex)
+        , _type(type)
+        , _value(value)
+        , _index(index)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags() | GA_FLAGS::ALLOW_WHILE_PAUSED;
+    }
+
+    void Serialise(DataSerialiser & stream) override
+    {
+        GameAction::Serialise(stream);
+        auto type = static_cast<uint8_t>(_type);
+        stream << DS_TAG(_rideIndex) << DS_TAG(type) << DS_TAG(_value) << DS_TAG(_index);
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        if (_rideIndex >= MAX_RIDES)
```

Comment:
Check for both MAX_RIDES and RIDE_ID_NULL.

Reply:
They are the same value. But i can certainly add an additional check if you think its necessary.

Reply:
Yes, please. They are the same value currently, but that won't necessarily always be true. Better to be explicit.